assertj-core/src/main/java/org/assertj/core/internal/BigDecimals.java

```diff
@@ -84,4 +85,10 @@ public void assertHasScale(AssertionInfo info, BigDecimal actual, int expectedSc
     throw failures.failure(info, shouldHaveScale(actual, expectedScale));
   }
 
+  public void assertHasPrecision(AssertionInfo info, BigDecimal actual, int expectedPrecision) {
+    assertNotNull(info, actual);
+    if (areEqual(actual.precision(), expectedPrecision)) return;
+    throw failures.failure(info, shouldHavePrecision(actual, expectedPrecision));
+  }
```

Comment:
I imagine you've taken inspiration from `assertHasScale`. However, these assertions are hardly reusable somewhere else so I suggest pulling this to `AbstractBigDecimalAssert`, similar to what we have for `AbstractCharSequenceAssert::isAlphabetic`:

https://github.com/assertj/assertj/blob/fd20abf93fd58a93f02ff9f1353dd6ae6e62f755/assertj-core/src/main/java/org/assertj/core/api/AbstractCharSequenceAssert.java#L2018-L2022